### PR TITLE
Fix #5038 (@wdio-allure-reporter) strip out ASCII codes from expect-webdriverio

### LINKED ISSUE #5038: Can't generate Allure report while test failed
**Environment (please complete the following information):**
 - **WebdriverIO version:** 5.18.7
 - **Mode:** WDIO Testrunner
 - **If WDIO Testrunner, running sync/async:** sync
 - **Node.js version:** 12.16.0
 - **NPM version:** 6.13.4
 - **Browser name and version:** Chrome 80.0.3987.106
 - **Platform name and version:** Windows 10
 - **Additional wdio packages used (if applicable):** @wdio/allure-reporter

**Config of WebdriverIO**
? Where should your tests be launched? local
? Where is your automation backend located? On my local machine
? Which framework do you want to use? cucumber
? Do you want to run WebdriverIO commands synchronous or asynchronous? sync
? Where are your feature files located? ./features/**/*.feature
? Where are your step definitions located? ./features/step-definitions
? Which reporter do you want to use? spec, allure
? Do you want to add a service to your test setup? chromedriver
? What is the base url? http://localhost

**Describe the bug**
While I run the test successfully, the allure-reporter can generate the xml file. But when the test failed,  the allure-reporter didn't generate any file so that I can't check the failed test result.

**To Reproduce**
The allure-reporter configure is
```=JAVASCRIPT
reporters: ['spec', ['allure', {
    outputDir: './allure-results',
    useCucumberStepReporter: true
 }]],
```
Then I run `wdio wdio.conf.js`.
***The test failed (This is what I expected), the result xml file not generated (This is not what I expected).***
Besides, I found some error message may related to this bug
```
[0-0] 2020-02-18T09:17:34.384Z ERROR @wdio/runner: Error: charData should not contain characters not allowed in XML
    at XmlCharData.set [as charData] (D:\c\Desktop\ProgramProject\HTML\Homework\HW-5\node_modules\xmlcreate\lib\nodes\XmlCharData.js:75:23)
    at new XmlCharData (D:\c\Desktop\ProgramProject\HTML\Homework\HW-5\node_modules\xmlcreate\lib\nodes\XmlCharData.js:53:24)
    at XmlElement.charData (D:\c\Desktop\ProgramProject\HTML\Homework\HW-5\node_modules\xmlcreate\lib\nodes\XmlElement.js:168:28)
    at parseString (D:\c\Desktop\ProgramProject\HTML\Homework\HW-5\node_modules\js2xmlparser\lib\main.js:53:27)
    at parseValue (D:\c\Desktop\ProgramProject\HTML\Homework\HW-5\node_modules\js2xmlparser\lib\main.js:227:5)
    at parseObjectOrMapEntry (D:\c\Desktop\ProgramProject\HTML\Homework\HW-5\node_modules\js2xmlparser\lib\main.js:132:5)
    at parseObjectOrMap (D:\c\Desktop\ProgramProject\HTML\Homework\HW-5\node_modules\js2xmlparser\lib\main.js:152:13)
    at parseValue (D:\c\Desktop\ProgramProject\HTML\Homework\HW-5\node_modules\js2xmlparser\lib\main.js:220:9)
    at parseObjectOrMapEntry (D:\c\Desktop\ProgramProject\HTML\Homework\HW-5\node_modules\js2xmlparser\lib\main.js:132:5)
    at parseObjectOrMap (D:\c\Desktop\ProgramProject\HTML\Homework\HW-5\node_modules\js2xmlparser\lib\main.js:152:13)
    at parseValue (D:\c\Desktop\ProgramProject\HTML\Homework\HW-5\node_modules\js2xmlparser\lib\main.js:220:9)
    at D:\c\Desktop\ProgramProject\HTML\Homework\HW-5\node_modules\js2xmlparser\lib\main.js:192:9
    at Array.forEach (<anonymous>)
    at parseArrayOrSet (D:\c\Desktop\ProgramProject\HTML\Homework\HW-5\node_modules\js2xmlparser\lib\main.js:187:16)
    at parseValue (D:\c\Desktop\ProgramProject\HTML\Homework\HW-5\node_modules\js2xmlparser\lib\main.js:224:9)
    at parseObjectOrMapEntry (D:\c\Desktop\ProgramProject\HTML\Homework\HW-5\node_modules\js2xmlparser\lib\main.js:132:5)
[0-0] Error: charData should not contain characters not allowed in XML
```
**Expected behavior**
When test failed, allure-reporter should also generate the result xml file.

**Log**
[Full log is here](https://gist.github.com/benebsiny/09bfa20b5a0afe3670e09005968ee679)
Sorry for containing some Chinese words in the log.

**Additional context**
Nope

> Please use the correct markdown

### PR DESCRIPTION
**Pull request by lacell75**
## Proposed changes

Fixed #5038 

## Types of changes

[//]: # (What types of changes does your code introduce to WebdriverIO?)
[//]: # (_Put an `x` in the boxes that apply_)

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## Checklist

[//]: # (_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._)

- [X] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/master/CONTRIBUTING.md) doc
- [X] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] I have added proper type definitions for new commands (if appropriate)

## Further comments

n/a

### CODE REVIEW

**Comment by christian-bromann** [comment]
@lacell75 thanks for proposing a PR!

@BorisOsipov what do you think? Should we rather move this patch to the Allure plugin?

**Comment by BorisOsipov** [comment]
@christian-bromann we use old allure js lib. In new versions they use json format for test results. Seems wdio fix will be enough.

**Comment by christian-bromann** [comment]
@BorisOsipov should we update to the new JS lib?

**Comment by BorisOsipov** [comment]
>should we update to the new JS lib? 

The allure generator supports both formats json and xml, I don't know are there any plans to get rid of xml format. But any way, we use old unsupported lib. We should update, but it is not easy - a new api is completly different from old :(


>Is the current beta version stable? 

Don't know. Seems yes - all new allure integrations use it.

File: packages/wdio-allure-reporter/src/utils.js [hunk 1]
```diff
@@ -1,7 +1,8 @@
 import process from 'process'
 import CompoundError from './compoundError'
 import { testStatuses, mochaEachHooks, mochaAllHooks, linkPlaceholder } from './constants'
-
+// eslint-disable-next-line no-control-regex
+const regex = new RegExp(/[\u001b\u009b][[()#;?]*(?:[0-9]{1,4}(?:;[0-9]{0,4})*)?[0-9A-ORZcf-nqry=><]/g)
```
Comment (christian-bromann) [inline packages/wdio-allure-reporter/src/utils.js:5]: can we comment where this regex comes from or is it build by you?

**Comment by lacell75** [comment]
Ok, I will use strip-ansi package.

File: packages/wdio-allure-reporter/src/utils.js (hunk 1 above)
Comment (lacell75) [reply packages/wdio-allure-reporter/src/utils.js]: I used ansi-strip instead of regex in new commit.

**Pull request merged**